core/audits/layout-shifts.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * @license Copyright 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {Audit} from './audit.js';
+import * as i18n from '../lib/i18n/i18n.js';
+import {CumulativeLayoutShift as CumulativeLayoutShiftComputed} from '../computed/metrics/cumulative-layout-shift.js';
+import CumulativeLayoutShift from './metrics/cumulative-layout-shift.js';
+import TraceElements from '../gather/gatherers/trace-elements.js';
+import {TraceEngineResult} from '../computed/trace-engine-result.js';
+
+/** @typedef {LH.Audit.Details.TableItem & {node?: LH.Audit.Details.NodeValue, score: number, subItems?: {type: 'subitems', items: SubItem[]}}} Item */
+/** @typedef {{node?: LH.Audit.Details.NodeValue, url?: string, cause: LH.IcuMessage}} SubItem */
+
+/* eslint-disable max-len */
+const UIStrings = {
+  /** Descriptive title of a diagnostic audit that provides the top elements affected by layout shifts. */
+  title: 'Avoid large layout shifts',
+  /** Description of a diagnostic audit that provides the top elements affected by layout shifts. "windowing" means the metric value is calculated using the subset of events in a small window of time during the run. "normalization" is a good substitute for "windowing". The last sentence starting with 'Learn' becomes link text to additional documentation. */
+  description: 'These are the largest layout shifts observed on the page. Some layout shifts may not be included in the CLS metric value due to [windowing](https://web.dev/articles/cls#what_is_cls). [Learn how to improve CLS](https://web.dev/articles/optimize-cls)',
+  /** Label for a column in a data table; entries in this column will be a number representing how large the layout shift was. */
+  columnScore: 'Layout shift score',
+  /** A possible reason why that the layout shift occured. */
+  rootCauseUnsizedMedia: 'A layout shift may have occurred because of CSS defining the size of an otherwise unsized element',
```

Comment:
I'd like to be more terse with these. Wdyt?

> Media element lacking an explicit size
> Web font loaded
> Injected iframe
> Network request loaded

All of these could have ", potentially triggering relayout"  on the end. Maybe we describe this in the description... (left a suggestion for that.)

I'm looking at the impl and go/ls-root-causes and think it may be tough to be more specific.

Reply:
>Suppose it really is just web font it catches.

ya. based on the impl it looks like its just webfont requests finishing around that time. you could almost consider "webfont loaded" a more specialized subset of "render blocking request loaded". (though i believe this renderblocking signal will not mark webfonts. so... it works out)